### ISSUE TITLE
Additional line breaks (or similar) when using nerdfonts on a windows machine with a linux shell (through WSL or ssh)

### ISSUE THREAD

**Issue by bmo-at** [ryanoasis/nerd-fonts#1246]
#### 🗹 Requirements

- [X] I have searched the [issues](https://github.com/ryanoasis/nerd-fonts/issues) for my issue and found nothing related and/or helpful
- [X] I have searched the [FAQ](https://github.com/ryanoasis/nerd-fonts/wiki/FAQ-and-Troubleshooting) for help
- [X] I have searched the [Wiki](https://github.com/ryanoasis/nerd-fonts/wiki) for help

#### 🎯 Subject of the issue

Experienced behavior:
When using a subset of nerdfonts icons (thus far it seems only to be nf-md-*), an additional line break and blank symbol will be inserted. This is suspected to be CR+LF from windows being interpreted by the linux shell as two line breaks, since linux only expects LF.

This does not occur und windows + powershell or linux + bash/zsh/etc.
It only occurs when using a linux shell under windows, e.g. via WSL or ssh. The terminal emulator seems to not matter, as this occurs both under Windows Terminal as well as Alacritty (for Windows).

Expected behavior:
There should be no line break after an icon.

Example symbols:
nf-md-battery 󰁹 f0079

#### 🔧 Your Setup

- _Which font are you using (e.g. `Anonymice Powerline Nerd Font Complete.ttf`)?_
  CaskaydiaCoveNerdFont-Regular.ttf
- _Where did you get the file from (download link, self patched, source downloaded from link...)_
   [nerdfonts download page](https://www.nerdfonts.com/font-downloads)
- _Which terminal emulator are you using (e.g. `iterm2`, `urxvt`, `gnome`, `konsole`)?_
   Windows Terminal
- _Are you using OS X, Linux or Windows? And which specific version or distribution?_
  Windows 11

#### ★ Screenshots (Optional)

Experienced:
![Screenshot 2023-05-11 125639](https://github.com/ryanoasis/nerd-fonts/assets/5568767/fcf51376-6b14-48db-93d8-c8835e77b389)

Expected: 
![Screenshot 2023-05-11 125523](https://github.com/ryanoasis/nerd-fonts/assets/5568767/70b66e74-30d9-4ee6-b744-916b5598ab7d)

**Comment by Finii** [comment]
Thanks for the Issue.
The font is installed on the Windows side, right?

**Comment by Finii** [comment]
Additional Idea:
Did you try outside the prompt, just `echo "xxx\UF0001xxx` (depending on your shell on Linux)?

**Comment by bmo-at** [comment]
> Thanks for the Issue. The font is installed on the Windows side, right?

Correct, version 3.0.1

> Additional Idea: Did you try outside the prompt, just `echo "xxx\UF0001xxx` (depending on your shell on Linux)?

```[bmo-at@PIKACHU ~]$ echo -e "xxx\Uf0079xxx"
xxx󰁹xxx
```
This seems to work fine

**Comment by bmo-at** [comment]
I also have this issue in my powerlevel10k prompt in zsh, so it seems to not just be an issue with my tide prompt on fish. Still trying to find a common thread between them

**Comment by Finii** [comment]
But the echo command shows that the handling itself is correct.
Probably it is
* the terminal cursor movement ansi sequences
* the length calculation in the prompt code

**Comment by TheSast** [comment]
Happends on Wezterm too.
This issue is prevelant when using tools such as tmux or zellij, it can make viewing or editing files inside the terminal that have any icons in them very hard.

**Comment by Finii** [comment]
@TheSast In this WSL situation, or on pure Windows resp pure Linux?

**Comment by TheSast** [comment]
Inside WSL

**Comment by bmo-at** [comment]
> * the length calculation in the prompt code
This I can pretty much say isn't it because the same prompt executing the same code produces different results on my server depending on wether I ssh into it via windows or look at the console locally via a connected monitor

> * the terminal cursor movement ansi sequences
This is the likely culprit, as I can also trigger some weird behavior with the nerdfonts icons and cursor placement when editing some string containing the icon:
When I type copy the icon from the nerdfonts website and paste it into bash, this happens:
![image](https://github.com/ryanoasis/nerd-fonts/assets/5568767/53ed2ef2-2158-4fee-850b-abb4e234dcb6)
When I hit backspace once, the space disappears but not the icon:
![image](https://github.com/ryanoasis/nerd-fonts/assets/5568767/b3756eee-f5a7-44fe-b400-c6d0a62d4424)
And finally, when I then type something, `clear` for example, it executes the command correctly, indicating that the backspace correctly deleted the entire icon:
![image](https://github.com/ryanoasis/nerd-fonts/assets/5568767/92c71c27-415d-4f78-a1ce-87ef971ab27b)

**Comment by Finii** [comment]
Time to use `zutty` as the only terminal that gets the sequences right? ;-D

https://tomscii.sig7.se/2020/12/A-totally-biased-comparison-of-Zutty

But then, if I understand correctly it does not happen when the same terminal application is used natively (i.e. without WSL), meaning it has something to do with **it**, WSL. 

This is like Wine. It might work, sometimes. With winetricks installed.

**Comment by Finii** [comment]
> It only occurs when using a linux shell under windows, e.g. via WSL or ssh. The terminal emulator seems to not matter

Also when ssh-ing to a regular Linux box with ssh from Windows? Then ssh must be broken. Is ssh also a windows command line application? Trying some open source ssh thing? Is not putty such a thing, terminal with integrated ssh?

**Comment by sirfz** [comment]
I think the issue is connected to tmux and not WSL/terminal? I use tmux on a remote machine and these "ghost" spaces are rendered with some icons.

Inside tmux:

![image](https://github.com/ryanoasis/nerd-fonts/assets/4741099/8cab2bf3-adca-4122-abf4-107446ceb49e)

Outside tmux:

![image](https://github.com/ryanoasis/nerd-fonts/assets/4741099/420bddf3-c98a-4eea-843b-b21bfb58c2bf)

**Comment by Finii** [comment]
@sirfz Thanks for showing.
And there is no Windows involved?
I think maybe additional the terminal is important, which VT tmux is using

I have never used WSL and have no clue how to set it up to test things. So a one-OS scenario would really help.

**Comment by sirfz** [comment]
This is on Windows 10, WSL, Windows Terminal. Issue only appears inside tmux (I only tested on a remote server connected via ssh).

tmux.conf:
```
# nvim recommended
set -g default-terminal "tmux-256color"
set -ag terminal-overrides ",xterm-256color:RGB"
set -sg escape-time 10
set -g focus-events on
# ctrl+arrows enabled (not needed?)
set-window-option -g xterm-keys on
```

**Comment by sirfz** [comment]
Alright, I can confirm that WSL certainly is causing the issue (which occurs when using tmux). I tried wezterm with WSL and was able to reproduce the issue.

However, when I connect using wezterm's mux server (which bypasses WSL), the issue did not appear.

**Comment by youngtuotuo** [comment]
This got solved when I switch to Windows Terminal Preview version (1.17.1023). 

But there's still minor different, as you can see in the screenshot.
Both use the same font and theme setting

- Theme: Kanagawa
- Font: FiraCode Nerd Font (no Mono)
- Fontsize: 10

![Screenshot 2023-05-23 202910](https://github.com/ryanoasis/nerd-fonts/assets/56819408/f5af0a0f-e4f2-4fdb-9a55-87aa904e1fbc)

**Comment by bmo-at** [comment]
@youngtuotuo Yeah I observed the same behavior when switching to the wt 1.17 preview. I'll link the issue from the microsoft/terminal repo, but that was closed for being a duplicate (they are aware of Unicode related issues and have a rewrite coming out in 1.18).
So right now I'm gonna wait for 1.18 to come out as a preview, and if the icons are still to small like they are in 1.17 I will open another issue related to the sizing. But other than them being to small, everything seems to work for me. Can you confirm that experience?

**Comment by bmo-at** [comment]
Ok so this really seems to be a Unicode support issue. Yesterday, Microsoft released Terminal Preview 1.18, with their new rendering engine which has better unicode support and no remaining issues. Icons are the right size and don't contain the line breaks. Closing this now, but for future reference: If this occurs it is likely your terminal thats at fault.

**Comment by Finii** [comment]
Thanks @bmo-at for following up and reporting the root cause :+1:
